src/Microsoft.DotNet.Darc/src/Darc/Operations/GetRepositoryMergePoliciesOperation.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Darc.Helpers;
+using Microsoft.DotNet.Darc.Options;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Darc.Operations
+{
+    internal class GetRepositoryMergePoliciesOperation : Operation
+    {
+        private GetRepositoryMergePoliciesCommandLineOptions _options;
+
+        public GetRepositoryMergePoliciesOperation(GetRepositoryMergePoliciesCommandLineOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        public override async Task<int> ExecuteAsync()
+        {
+            try
+            {
+                IRemote remote = RemoteFactory.GetBarOnlyRemote(_options, Logger);
+
+                IEnumerable<RepositoryBranch> allRepositories = await remote.GetRepositoriesAsync();
+                IEnumerable<RepositoryBranch> filteredRepositories = allRepositories.Where(repositories =>
+                    (string.IsNullOrEmpty(_options.Repo) || repositories.Repository.Contains(_options.Repo, StringComparison.OrdinalIgnoreCase)) &&
+                    (string.IsNullOrEmpty(_options.Branch) || repositories.Branch.Contains(_options.Branch, StringComparison.OrdinalIgnoreCase)));
+
+                // List only those repos and branches that are targeted by a batchable subscription (active) unless the user
+                // passes --all.
+                if (!_options.All)
+                {
+                    HashSet<string> batchableTargets = (await remote.GetSubscriptionsAsync())
+                        .Where(s => s.Policy.Batchable)
+                        .Select<Subscription, string>(s => $"{s.TargetRepository}{s.TargetBranch}")
```

Comment:
[nit] Would a  (string repo, string branch) tuple be better than building this string?

Reply:
I like this methodology because a case insensitive comparison is simple vs. a tuple comparison (which would require implementing `IEqualityComparer`). If this HashSet was used in a larger scope that would be a better idea for sure.